File: day 2 Java, Rust, Go factorial/prvocisla.go
package main

import "fmt"

func main() {


 fmt.Printf("Vitej, jdeme hledat prvočísla. GL !!");
 fmt.Printf("%c", '\n');

  var aktdelka int = 0;
	var cisla[10001]int;
	cisla[0] = 2 
  var ccislo = 2;

  for i := 0; i < 10000; i++ {
    var ok = false;
    for !ok {
      ccislo++;
      for  i := 0; i <= aktdelka; i++ {
        if (ccislo % cisla[i] != 0) {
          ok = true;
        } else{
          ok = false;
          break;
        }
      }
    }
    aktdelka++;
    cisla[aktdelka] = ccislo;
  }

  for i := 0; i < aktdelka; i++ {
   fmt.Print(cisla[i], ", ");
  }

	fmt.Printf("\n konec %c", '\n');


}

File: day 2 Go/prvocisla.c


#include <stdio.h>
#include <stdbool.h>

int main()
{
  printf("Vitej, jdeme hledat prvočísla. GL !!");

  printf("%c", '\n');

  int aktdelka = 0;
  unsigned long cisla[1000] = {2};
  unsigned long ccislo = 2;

  for (int i = 0; i < 1000; i++)
  {
    bool ok = false;
    while (!ok)
    {
      ccislo++;

      // printf("trying: %d \n", ccislo);
      for (int i = 0; i <= aktdelka; i++)
      {

        if (ccislo % cisla[i] != 0)
        {
          // printf("success: %d \n", ccislo);
          ok = true;
        }
        else
        {
          ok = false;
          break;
        }
      }
    }
    aktdelka++;
    cisla[aktdelka] = ccislo;
  }

  for (int i = 0; i < aktdelka; i++)
  {
    printf("%ld, ", cisla[i]);
  }

  printf("\n konec %c", '\n');

  return 0;
}